### DISCUSSION TITLE
events start after 86400

### DISCUSSION THREAD

**Discussion by syleemy** [matsim-org/matsim-episim-libs#71] [Q&A]
Hi again,

May I confirm the way Episim handles these situations:
1. The `time `for `actstart `for a person is > 86400. 
It seems that Episim forces the end of day to be 86400. So, this person will still be added to the container, but maybe no infection will happen to him?

2. There is no corresponding `actend `after `actstart`. e.g., 
![1](https://user-images.githubusercontent.com/51136409/114116940-ead9b100-98e5-11eb-98c0-5113f46501ab.jpg)
Will this person be teleported back to home at end of the day?
The current `addSpentTime `in `EpisimPerson `cannot calculate the spent time for the last event at `13938`.

Thank you in advance again.

**Comment by rakow** [comment]
Hello,

1. Contacts between persons are still evaluated (on actend), even if it is greater 86400. Infection time will be rounded to 86400, though.
2. Yes, people with unclosed trajectories are put into their first container at the end of the day (usually home). In your example the person performs leisure for the rest of the day. The `spentTime` should also be updated under this assumption, but I may have to look at it again.

**Reply by syleemy** [reply]
Hi rakow,

Thanks for the prompt reply. Here I write my observations that could be useful for others after running some experiments:
1. The jointTimeInContainer will always take a cutoff at `86400`. Example:
- person A enters FacilityA (`actstart`) at `80000`, leaves at `88000` (`actend`). The duration in FacilityA = 6400
- person B enters FacilityA at `88000`, leaves at `90000`. Duration = 0
- person C enters FacilityA at `89000`, leaves at `91000`. Duration = 0
The jointTimeinContainer between person B and person C in Facility A= 0. 

2. Because of characteristics in 1, some `SpentTime`  in `EpisimPerson` is = 0 for `actstart` that is after 86400

Thanks again!
